packs/icinga2/actions/lib/client.py

```diff
@@ -0,0 +1,128 @@
+import pycurl, json
+from StringIO import StringIO
+import urllib
+import signal
+import sys
+import time
+
+POST_PARAMS = {}
+MAX_ATTEMPTS = 3
+
+class Client:
+  def __init__(self, action, api_url, api_user, api_password, method):
+    self.conn = None
+    self.action = action
+    self.keep_trying = 1
+    self.api_url = api_url
+    self.api_user = api_user
+    self.api_password = api_password
+    self.method = method
+    self.buffer = ''
+    signal.signal(signal.SIGINT, self.handle_ctrl_c)
+
+  def handle_ctrl_c(self, signal, frame):
+    sys.stderr.write('SIGINT receivied')
+    sys.stderr.write('You pressed Ctrl+C?!')
+    self.abort_session()
+
+  def abort_session(self):
+    self.keep_trying = 0
+    self.conn.close()
+
+  def setup_connection(self):
+
+    if self.conn:
+      self.conn.close()
+    self.headers = ['Accept: application/json', 'Expect:', 'Connection: close']
+    self.conn = pycurl.Curl()
```

Comment:
Any particular reason why you used pycurl and not requests?

Requests is much simpler and requires a lot less boiler-plate code - http://docs.python-requests.org/en/master/